### PR TITLE
upcoming: [M3-7874] - Linode Create Refactor - Marketplace App Sections

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Adds the `New apps`, `Popular apps`, and `all apps` sections to the new Linode Create flow 🆕 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-05-28 at 3 28 15 PM | Screenshot 2024-05-28 at 3 28 03 PM |
| Notice how there are no sections  | It now looks like the old flow with the proper sections  | 

### Prerequisites
- Turn on the Linode Create v2 feature flag using local dev tools

### Verification steps
- Checkout this PR locally or use the internal preview link
- Go to the Linode Create flow
- Got to the marketplace tab
- Verify you see the sections `New apps`, `Popular apps`, and `all apps` 
- Check for parity with production (the old Linode Create flow)
  - Verify the apps are the same under each section

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/OneClickApps/types.ts [hunk 1]
```diff
@@ -5,6 +5,13 @@ export interface OCA {
   colors: Colors;
   description: string;
   href?: string;
+  /**
+   * Set isNew to `true` if you want the app to show up in the "New apps"
+   * section on the Linode Create flow.
+   *
+   * @note this value only affects Linode Create v2
+   */
+  isNew?: boolean;
```
[the remark this answers is not in the document]
Comment (hkhalil-akamai) [reply packages/manager/src/features/OneClickApps/types.ts:14]: If the new apps are rotated frequently, maybe its better to use a feature flag? (So it's not dependent on our release cycle)

**Review by hkhalil-akamai — APPROVED** [review]
Code changes are looking clean 🧼 and found no regressions from prod.

File: packages/manager/src/features/OneClickApps/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/OneClickApps/types.ts:14]: Yeah, this is a good point.  Before we launch v2, I will make a new flag that will allow us to override **anything** for emergencies

**Review by jaalah-akamai — APPROVED** [review]

**Review by hana-linode — COMMENTED** [review]
Can we also fix the Setup headers for special characters and/or add back the app icon & node notice?

| Prod  | Linode Create v2   |
| ------- | ------- |
| image | image |

**Comment by bnussman-akamai** [comment]
@hana I addressed that feedback!

I also went ahead and deleted the old `oneClickApps` list and the `stackScriptUtils.ts` baseApps list. `oneClickAppsv2` is now the source of truth.

**Comment by bnussman-akamai** [comment]
@hana-linode Search is not yet implemented, it will come in a later PR

**Review by hana-linode — APPROVED** [review]

**Pull request merged**